--- a/Makefile	2020-07-21 11:42:38.277872138 +0800
+++ b/Makefile	2020-07-21 11:41:15.738509033 +0800
@@ -47,7 +47,7 @@
 ifeq ($(MT7621_MP), y)
 CONFIG_CROSS_COMPILER_PATH ?= /opt/mips-2012.03/bin/
 else
-CONFIG_CROSS_COMPILER_PATH ?= /opt/buildroot-gcc342/bin
+CONFIG_CROSS_COMPILER_PATH ?= $(TOPDIR)/toolchain_opt/buildroot-gcc342/bin
 endif
 
 ifeq (include/config.mk,$(wildcard include/config.mk))
@@ -1663,6 +1663,33 @@
 
 rt2880_config: unconfig
 	@./mkconfig $(@:_config=) mips ralink_soc rt2880
+
+widora_neo_32m_config:
+	echo "Fake config for Widora Neo 32M"
+kooiot_tlink_c1_config:
+	echo "Fake config for ThingsLink C1"
+kooiot_tlink_c1_16m_config:
+	echo "Fake config for ThingsLink C1 (16M)"
+kooiot_tlink_c1_32m_config:
+	echo "Fake config for ThingsLink C1 (32M)"
+kooiot_tlink_c2_config:
+	echo "Fake config for ThingsLink C2"
+kooiot_tlink_c2_16m_config:
+	echo "Fake config for ThingsLink C2 (16M)"
+kooiot_tlink_c2_32m_config:
+	echo "Fake config for ThingsLink C2 (32M)"
+kooiot_tlink_c2_v2_config:
+	echo "Fake config for ThingsLink C2 V2 (32M)"
+kooiot_tlink_c3_config:
+	echo "Fake config for ThingsLink C3"
+kooiot_tlink_c3_16m_config:
+	echo "Fake config for ThingsLink C3 (16M)"
+kooiot_tlink_c3_32m_config:
+	echo "Fake config for ThingsLink C3 (32M)"
+kooiot_tlink_c4_16m_config:
+	echo "Fake config for ThingsLink C4 (16M)"
+kooiot_tlink_c4_32m_config:
+	echo "Fake config for ThingsLink C4 (32M)"
 #########################################################################
 ## MIPS32 AU1X00
 #########################################################################
