diff --git a/configure.ac b/configure.ac
index 42ae5adcf..53dfdf305 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1477,23 +1477,7 @@ if test "x$ss_cv_static_plugin_constructors" = xyes; then
 	fi
 fi
 
-AC_MSG_CHECKING([version from Git repository])
-AC_SUBST(GIT_VERSION, [$($srcdir/scripts/git-version "$srcdir")])
-case "$GIT_VERSION" in
-	"$PACKAGE_VERSION"*)
-		AC_MSG_RESULT([$GIT_VERSION])
-		;;
-	UNKNOWN)
-		AC_MSG_RESULT([$GIT_VERSION (could be due to insufficient clone depth)])
-		;;
-	*)
-		AC_MSG_ERROR([$PACKAGE_VERSION is not a prefix of $GIT_VERSION, tag missing?])
-		;;
-esac
-
-if test x$git_version = xtrue -a "$GIT_VERSION" != "UNKNOWN"; then
-	AC_DEFINE_UNQUOTED(VERSION, ["$GIT_VERSION"])
-fi
+AC_DEFINE_UNQUOTED(VERSION, ["$PACKAGE_VERSION"])
 
 # modify CFLAGS as needed, do this late so we don't affect configure checks
 CFLAGS="$CFLAGS -include $(pwd)/config.h"
