--- a/include/command.h
+++ b/include/command.h
@@ -231,7 +231,7 @@ enum command_ret_t {
  *			number of ticks the command took to complete.
  * @return 0 if the command succeeded, 1 if it failed
  */
-int cmd_process(int flag, int argc, char *const argv[], int *repeatable,
+enum command_ret_t cmd_process(int flag, int argc, char *const argv[], int *repeatable,
 		unsigned long *ticks);

 void fixup_cmdtable(struct cmd_tbl *cmdtp, int size);
--- a/common/command.c
+++ b/common/command.c
@@ -584,7 +584,7 @@ static int cmd_call(struct cmd_tbl *cmdt
 }
 
 enum command_ret_t cmd_process(int flag, int argc, char *const argv[],
-			       int *repeatable, ulong *ticks)
+			       int *repeatable, unsigned long *ticks)
 {
 	enum command_ret_t rc = CMD_RET_SUCCESS;
 	struct cmd_tbl *cmdtp;
