--- a/aic8800_fdrv/rwnx_main.c
+++ b/aic8800_fdrv/rwnx_main.c
@@ -525,7 +525,8 @@ static const int rwnx_hwq2uapsd[NL80211_
 struct semaphore aicwf_deinit_sem;
 atomic_t aicwf_deinit_atomic;
 
-int aicwf_dbg_level = LOGERROR|LOGINFO|LOGTRACE|LOGDEBUG;
+// int aicwf_dbg_level = LOGERROR|LOGINFO|LOGTRACE|LOGDEBUG;
+int aicwf_dbg_level = LOGERROR|LOGINFO;
 module_param(aicwf_dbg_level, int, 0660);
 
 int testmode = 0;
--- a/aic_load_fw/aic_bluetooth_main.c
+++ b/aic_load_fw/aic_bluetooth_main.c
@@ -20,7 +20,8 @@ int adap_test = 0;
 char paringid[100];
 int n_para = 1;
 int ble_scan_wakeup_reboot_time = 1000;
-int aicwf_dbg_level = LOGERROR|LOGINFO|LOGDEBUG|LOGTRACE;
+// int aicwf_dbg_level = LOGERROR|LOGINFO|LOGDEBUG|LOGTRACE;
+int aicwf_dbg_level = LOGERROR|LOGINFO;
 int flash_erase_len = 0x400000;
 uint32_t ad_data_filter_mask = 0;
 uint32_t gpio_num = 2;//default select gpiob2 for fw_wakeup_host
