diff --git a/aic_load_fw/aicbluetooth.c b/aic_load_fw/aicbluetooth.c
index 0270623..d557a1d 100644
--- a/aic_load_fw/aicbluetooth.c
+++ b/aic_load_fw/aicbluetooth.c
@@ -169,7 +169,7 @@ enum aicbsp_cpmode_type {
 
 #define FW_PATH_MAX 200
 #if defined(CONFIG_PLATFORM_UBUNTU)
-static const char* aic_default_fw_path = "/lib/firmware";
+static const char* aic_default_fw_path = "/lib/firmware/aic8800_fw/USB";
 #else
 static const char* aic_default_fw_path = "/vendor/etc/firmware";
 #endif
