diff -rup aic8800_orig/aic8800_fdrv/rwnx_radar.c aic8800/aic8800_fdrv/rwnx_radar.c
--- aic8800_orig/aic8800_fdrv/rwnx_radar.c	2025-02-27 16:08:31.468795547 +0800
+++ aic8800/aic8800_fdrv/rwnx_radar.c	2025-02-27 16:16:04.412577146 +0800
@@ -1502,7 +1502,7 @@ static void rwnx_radar_cac_work(struct w
                     #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
                        &ctxt->chan_def,
                     #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)) || (BUILD_OPENWRT >= 2410)
                        NL80211_RADAR_CAC_FINISHED, GFP_KERNEL, 0);
 #else
                        NL80211_RADAR_CAC_FINISHED, GFP_KERNEL);
@@ -1644,7 +1644,7 @@ void rwnx_radar_cancel_cac(struct rwnx_r
                         #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
                            &ctxt->chan_def,
                         #endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0)) || (BUILD_OPENWRT >= 2410)
                            NL80211_RADAR_CAC_FINISHED, GFP_KERNEL, 0);
 #else
                            NL80211_RADAR_CAC_ABORTED, GFP_KERNEL);
diff -rup aic8800_orig/aic8800_fdrv/rwnx_main.c aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/aic8800_fdrv/rwnx_main.c	2025-02-27 16:23:10.358193938 +0800
+++ aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 16:24:06.665715638 +0800
@@ -1097,7 +1097,7 @@ static void rwnx_csa_finish(struct work_
         } else
             rwnx_txq_vif_stop(vif, RWNX_TXQ_STOP_CHAN, rwnx_hw);
         spin_unlock_bh(&rwnx_hw->cb_lock);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410))
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0);
 #elif (LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION3)
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0, 0);
@@ -4039,7 +4039,7 @@ static int rwnx_cfg80211_start_ap(struct
  * @change_beacon: Change the beacon parameters for an access point mode
  *	interface. This should reject the call when AP mode wasn't started.
  */
-#if LINUX_VERSION_CODE > KERNEL_VERSION(6, 7, 0)
+#if LINUX_VERSION_CODE > KERNEL_VERSION(6, 7, 0) || (BUILD_OPENWRT >= 2410)
 static int rwnx_cfg80211_change_beacon(struct wiphy *wiphy, struct net_device *dev,
                                        struct cfg80211_ap_update *info)
 #else
@@ -4059,7 +4059,7 @@ static int rwnx_cfg80211_change_beacon(s
 	elem.dma_addr = 0;

     // Build the beacon
-#if LINUX_VERSION_CODE > KERNEL_VERSION(6, 7, 0)
+#if LINUX_VERSION_CODE > KERNEL_VERSION(6, 7, 0) || (BUILD_OPENWRT >= 2410)
     buf = rwnx_build_bcn(bcn, &info->beacon);
 #else
     buf = rwnx_build_bcn(bcn, info);
@@ -4171,6 +4171,9 @@ static int rwnx_cfg80211_set_monitor_cha
                                              struct cfg80211_chan_def *chandef)
 #else
 static int rwnx_cfg80211_set_monitor_channel(struct wiphy *wiphy,
+#if (BUILD_OPENWRT >= 2410)
+											struct net_device *,
+#endif
                                              struct cfg80211_chan_def *chandef)
 #endif
 {
@@ -4909,7 +4912,10 @@ static
 int rwnx_cfg80211_start_radar_detection(struct wiphy *wiphy,
                                         struct net_device *dev,
                                         struct cfg80211_chan_def *chandef
-                                    #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
+									#if (BUILD_OPENWRT >= 2410)
+										, u32 cac_time_ms
+										, int link_id
+                                    #elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
                                         , u32 cac_time_ms
                                     #endif
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 0))
@@ -5066,7 +5072,7 @@ int rwnx_cfg80211_channel_switch(struct
         goto end;
     } else {
         INIT_WORK(&csa->work, rwnx_csa_finish);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410)
 	cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false);
 #elif LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION4
 	cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false, 0);
diff -rup aic8800_orig/aic8800_fdrv/rwnx_main.c aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/aic8800_fdrv/rwnx_main.c	2025-02-27 16:23:10.358193938 +0800
+++ aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 16:24:06.665715638 +0800
@@ -4240,7 +4240,11 @@ int rwnx_cfg80211_set_monitor_channel_(s
 int rwnx_cfg80211_set_monitor_channel_(struct wiphy *wiphy,
                                              struct cfg80211_chan_def *chandef)
 {
+#if (BUILD_OPENWRT >= 2410)
+    return rwnx_cfg80211_set_monitor_channel(wiphy, NULL, chandef);
+#else
     return rwnx_cfg80211_set_monitor_channel(wiphy, chandef);
+#endif
 }
 #endif

@@ -4721,7 +4721,7 @@ static int rwnx_cfg80211_get_channel(str
     if (rwnx_vif->vif_index == rwnx_hw->monitor_vif)
     {
         //retrieve channel from firmware
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 13, 0)) || (BUILD_OPENWRT >= 2410)
         rwnx_cfg80211_set_monitor_channel(wiphy, wdev->netdev, NULL);
 #else
         rwnx_cfg80211_set_monitor_channel(wiphy, NULL);
