diff -rup aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 11:41:29.405135319 +0800
+++ aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 11:47:21.790647983 +0800
@@ -783,8 +783,8 @@ static void rwnx_csa_finish(struct work_
 	} else {
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
 		mutex_lock(&vif->wdev.mtx);
-#endif
 		__acquire(&vif->wdev.mtx);
+#endif
 		spin_lock_bh(&rwnx_hw->cb_lock);
 		rwnx_chanctx_unlink(vif);
 		rwnx_chanctx_link(vif, csa->ch_idx, &csa->chandef);
@@ -805,8 +805,8 @@ static void rwnx_csa_finish(struct work_
 #endif
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
 		mutex_unlock(&vif->wdev.mtx);
-#endif
 		__release(&vif->wdev.mtx);
+#endif
 	}
 	rwnx_del_csa(vif);
 }
diff -rup aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_rx.c aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_rx.c
--- aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_rx.c	2025-02-27 11:41:29.413135450 +0800
+++ aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_rx.c	2025-02-27 11:42:40.778290673 +0800
@@ -456,7 +456,7 @@ static bool rwnx_rx_data_skb(struct rwnx
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 3, 0)
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 4, 0)
 		ieee80211_amsdu_to_8023s(skb, &list, rwnx_vif->ndev->dev_addr,
-								 RWNX_VIF_TYPE(rwnx_vif), 0, NULL, NULL, NULL);
+								 RWNX_VIF_TYPE(rwnx_vif), 0, NULL, NULL, 0);
 #else
 		ieee80211_amsdu_to_8023s(skb, &list, rwnx_vif->ndev->dev_addr,
 								RWNX_VIF_TYPE(rwnx_vif), 0, NULL, NULL, false);
diff -rup aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 11:41:29.389135057 +0800
+++ aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 11:43:47.815355396 +0800
@@ -1063,8 +1063,8 @@ static void rwnx_csa_finish(struct work_
     } else {
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
         mutex_lock(&vif->wdev.mtx);
-#endif
         __acquire(&vif->wdev.mtx);
+#endif
         spin_lock_bh(&rwnx_hw->cb_lock);
         rwnx_chanctx_unlink(vif);
         rwnx_chanctx_link(vif, csa->ch_idx, &csa->chandef);
@@ -1085,8 +1085,8 @@ static void rwnx_csa_finish(struct work_
 #endif
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
         mutex_unlock(&vif->wdev.mtx);
-#endif
         __release(&vif->wdev.mtx);
+#endif
     }
     rwnx_del_csa(vif);
 }
diff -rup aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c
--- aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c	2025-02-27 11:41:29.389135057 +0800
+++ aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_rx.c	2025-02-27 11:42:03.677693096 +0800
@@ -467,7 +467,7 @@ static bool rwnx_rx_data_skb(struct rwnx
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 3, 0)
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 4, 0)
         ieee80211_amsdu_to_8023s(skb, &list, rwnx_vif->ndev->dev_addr,
-                                 RWNX_VIF_TYPE(rwnx_vif), 0, NULL, NULL, NULL);
+                                 RWNX_VIF_TYPE(rwnx_vif), 0, NULL, NULL, 0);
 #else
         ieee80211_amsdu_to_8023s(skb, &list, rwnx_vif->ndev->dev_addr,
                                  RWNX_VIF_TYPE(rwnx_vif), 0, NULL, NULL, false);
diff -rup aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:17:16.328450446 +0800
+++ aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:19:25.934763380 +0800
@@ -781,7 +781,7 @@ static void rwnx_csa_finish(struct work_
 		cfg80211_disconnected(vif->ndev, 0, NULL, 0, 0, GFP_KERNEL);
 		#endif
 	} else {
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
+#if ((LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)) && (BUILD_OPENWRT < 2410))
 		mutex_lock(&vif->wdev.mtx);
 		__acquire(&vif->wdev.mtx);
 #endif
@@ -803,7 +803,7 @@ static void rwnx_csa_finish(struct work_
 #else
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef);
 #endif
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
+#if ((LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)) && (BUILD_OPENWRT < 2410))
 		mutex_unlock(&vif->wdev.mtx);
 		__release(&vif->wdev.mtx);
 #endif
diff -rup aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:17:16.292449797 +0800
+++ aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:18:30.713783933 +0800
@@ -1061,7 +1061,7 @@ static void rwnx_csa_finish(struct work_
         cfg80211_disconnected(vif->ndev, 0, NULL, 0, 0, GFP_KERNEL);
         #endif
     } else {
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
+#if ((LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)) && (BUILD_OPENWRT < 2410))
         mutex_lock(&vif->wdev.mtx);
         __acquire(&vif->wdev.mtx);
 #endif
@@ -1083,7 +1083,7 @@ static void rwnx_csa_finish(struct work_
 #else
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef);
 #endif
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
+#if ((LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)) && (BUILD_OPENWRT < 2410))
         mutex_unlock(&vif->wdev.mtx);
         __release(&vif->wdev.mtx);
 #endif
diff -rup aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:52:36.499680214 +0800
+++ aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:53:14.857701836 +0800
@@ -1074,7 +1074,7 @@ static void rwnx_csa_finish(struct work_
         } else
             rwnx_txq_vif_stop(vif, RWNX_TXQ_STOP_CHAN, rwnx_hw);
         spin_unlock_bh(&rwnx_hw->cb_lock);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410))
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0);
 #elif (LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION3)
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0, 0);
@@ -6059,7 +6059,7 @@ int rwnx_cfg80211_channel_switch(struct
         goto end;
     } else {
         INIT_WORK(&csa->work, rwnx_csa_finish);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410))
 		cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false);
 #elif LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION4
 		cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false, 0);
diff -rup aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:52:36.523668375 +0800
+++ aic8800/PCIE/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 15:54:28.907122116 +0800
@@ -794,7 +794,7 @@ static void rwnx_csa_finish(struct work_
 		} else
 			rwnx_txq_vif_stop(vif, RWNX_TXQ_STOP_CHAN, rwnx_hw);
 		spin_unlock_bh(&rwnx_hw->cb_lock);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410))
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0);
 #elif (LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION3)
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0, 0);
@@ -5811,7 +5811,7 @@ int rwnx_cfg80211_channel_switch(struct
         rwnx_del_csa(vif);
     } else {
         INIT_WORK(&csa->work, rwnx_csa_finish);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410))
 		cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false);
 #elif LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION4
 		cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false, 0);
diff -rup aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 16:08:31.468795547 +0800
+++ aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 16:14:05.717029360 +0800
@@ -5084,7 +5084,7 @@ static int rwnx_cfg80211_start_ap(struct
  * @change_beacon: Change the beacon parameters for an access point mode
  *	interface. This should reject the call when AP mode wasn't started.
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)) || (BUILD_OPENWRT >= 2410))
 static int rwnx_cfg80211_change_beacon(struct wiphy *wiphy, struct net_device *dev,
                                        struct cfg80211_ap_update *params)
 #else
@@ -5201,6 +5201,9 @@ cfg80211_chandef_identical(const struct
 #endif
 
 static int rwnx_cfg80211_set_monitor_channel(struct wiphy *wiphy,
+#if (BUILD_OPENWRT >= 2410)
+											struct net_device *,
+#endif
                                              struct cfg80211_chan_def *chandef)
 {
     struct rwnx_hw *rwnx_hw = wiphy_priv(wiphy);
@@ -5907,7 +5910,10 @@ static
 int rwnx_cfg80211_start_radar_detection(struct wiphy *wiphy,
                                         struct net_device *dev,
                                         struct cfg80211_chan_def *chandef
-                                    #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
+									#if (BUILD_OPENWRT >= 2410)
+										, u32 cac_time_ms
+										, int link_id
+                                    #elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
                                         , u32 cac_time_ms
                                     #endif
                                         )
diff -rup aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_radar.c aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_radar.c
--- aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_radar.c	2025-02-27 16:08:31.468795547 +0800
+++ aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_radar.c	2025-02-27 16:16:04.412577146 +0800
@@ -1395,11 +1395,15 @@ static void rwnx_radar_cac_work(struct w
     }
 
     ctxt = &rwnx_hw->chanctx_table[radar->cac_vif->ch_index];
-    cfg80211_cac_event(radar->cac_vif->ndev,
+    cfg80211_cac_event(radar->cac_vif->ndev,	
                     #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
                        &ctxt->chan_def,
                     #endif
+					#if BUILD_OPENWRT >= 2410
+                       NL80211_RADAR_CAC_FINISHED, GFP_KERNEL, 0);
+					#else
                        NL80211_RADAR_CAC_FINISHED, GFP_KERNEL);
+					#endif
     rwnx_send_apm_stop_cac_req(rwnx_hw, radar->cac_vif);
     rwnx_chanctx_unlink(radar->cac_vif);
 
@@ -1499,7 +1503,11 @@ void rwnx_radar_cancel_cac(struct rwnx_r
                         #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
                            &ctxt->chan_def,
                         #endif
+					#if BUILD_OPENWRT >= 2410
+                           NL80211_RADAR_CAC_ABORTED, GFP_KERNEL, 0);
+					#else
                            NL80211_RADAR_CAC_ABORTED, GFP_KERNEL);
+					#endif
         rwnx_chanctx_unlink(radar->cac_vif);
     }
 
diff -rup aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 16:23:10.358193938 +0800
+++ aic8800/USB/driver_fw/drivers/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 16:24:06.665715638 +0800
@@ -5104,7 +5104,7 @@ static int rwnx_cfg80211_change_beacon(s
 	elem.dma_addr = 0;
 
     // Build the beacon
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)) || (BUILD_OPENWRT >= 2410)
     buf = rwnx_build_bcn(bcn, &params->beacon);
 #else
     buf = rwnx_build_bcn(bcn, info);
@@ -5729,7 +5729,7 @@ static int rwnx_cfg80211_get_channel(str
     if (rwnx_vif->vif_index == rwnx_hw->monitor_vif)
     {
         //retrieve channel from firmware
-        rwnx_cfg80211_set_monitor_channel(wiphy, NULL);
+        rwnx_cfg80211_set_monitor_channel(wiphy, NULL, NULL);
     }
 
     //Check if channel context is valid
diff -rup aic8800_orig/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c aic8800/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c
--- aic8800_orig/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 17:09:52.379749925 +0800
+++ aic8800/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_main.c	2025-02-27 17:15:37.012627518 +0800
@@ -770,10 +770,10 @@ static void rwnx_csa_finish(struct work_
 		cfg80211_disconnected(vif->ndev, 0, NULL, 0, 0, GFP_KERNEL);
 		#endif
 	} else {
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)) && (BUILD_OPENWRT < 2410)
 		mutex_lock(&vif->wdev.mtx);
-#endif
 		__acquire(&vif->wdev.mtx);
+#endif
 		spin_lock_bh(&rwnx_hw->cb_lock);
 		rwnx_chanctx_unlink(vif);
 		rwnx_chanctx_link(vif, csa->ch_idx, &csa->chandef);
@@ -783,7 +783,7 @@ static void rwnx_csa_finish(struct work_
 		} else
 			rwnx_txq_vif_stop(vif, RWNX_TXQ_STOP_CHAN, rwnx_hw);
 		spin_unlock_bh(&rwnx_hw->cb_lock);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410))
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0);
 #elif (LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION3)
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef, 0, 0);
@@ -792,10 +792,10 @@ static void rwnx_csa_finish(struct work_
 #else
 		cfg80211_ch_switch_notify(vif->ndev, &csa->chandef);
 #endif
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 7, 0)) && (BUILD_OPENWRT < 2410)
 		mutex_unlock(&vif->wdev.mtx);
-#endif
 		__release(&vif->wdev.mtx);
+#endif
 	}
 	rwnx_del_csa(vif);
 }
@@ -4759,7 +4759,7 @@ end:
  * @change_beacon: Change the beacon parameters for an access point mode
  *	interface. This should reject the call when AP mode wasn't started.
  */
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)) || (BUILD_OPENWRT >= 2410))
 static int rwnx_cfg80211_change_beacon(struct wiphy *wiphy, struct net_device *dev,
 									   struct cfg80211_ap_update *params)
 #else
@@ -4778,7 +4778,7 @@ static int rwnx_cfg80211_change_beacon(s
 	RWNX_DBG(RWNX_FN_ENTRY_STR);
 
 	// Build the beacon
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0)) || (BUILD_OPENWRT >= 2410)
 	buf = rwnx_build_bcn(bcn, &params->beacon);
 #else
 	buf = rwnx_build_bcn(bcn, info);
@@ -4850,6 +4850,9 @@ static int rwnx_cfg80211_stop_ap(struct
  * configured at firmware level.
  */
 static int rwnx_cfg80211_set_monitor_channel(struct wiphy *wiphy,
+#if (BUILD_OPENWRT >= 2410)
+		struct net_device *,
+#endif
 											 struct cfg80211_chan_def *chandef)
 {
 	struct rwnx_hw *rwnx_hw = wiphy_priv(wiphy);
@@ -5298,7 +5301,11 @@ static int rwnx_cfg80211_get_channel(str
 
 	if (rwnx_vif->vif_index == rwnx_hw->monitor_vif) {
 		//retrieve channel from firmware
+#if (BUILD_OPENWRT >= 2410)
+		rwnx_cfg80211_set_monitor_channel(wiphy, NULL, NULL);
+#else
 		rwnx_cfg80211_set_monitor_channel(wiphy, NULL);
+#endif
 	}
 
 	//Check if channel context is valid
@@ -5432,7 +5439,10 @@ static
 int rwnx_cfg80211_start_radar_detection(struct wiphy *wiphy,
 										struct net_device *dev,
 										struct cfg80211_chan_def *chandef
-									#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
+									#if (BUILD_OPENWRT >= 2410)
+										, u32 cac_time_ms
+										, int link_id
+									#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
 										, u32 cac_time_ms
 									#endif
 										)
@@ -5578,7 +5588,7 @@ int rwnx_cfg80211_channel_switch (struct
 		goto end;
 	} else {
 		INIT_WORK(&csa->work, rwnx_csa_finish);
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0))
+#if ((LINUX_VERSION_CODE >= KERNEL_VERSION(6, 9, 0)) || (BUILD_OPENWRT >= 2410))
 		cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false);
 #elif LINUX_VERSION_CODE >= HIGH_KERNEL_VERSION4
 		cfg80211_ch_switch_started_notify(dev, &csa->chandef, 0, params->count, false, 0);
diff -rup aic8800_orig/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_radar.c aic8800/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_radar.c
--- aic8800_orig/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_radar.c	2025-02-27 17:09:52.387750138 +0800
+++ aic8800/SDIO/driver_fw/driver/aic8800/aic8800_fdrv/rwnx_radar.c	2025-02-27 17:14:55.672716366 +0800
@@ -1399,7 +1399,11 @@ static void rwnx_radar_cac_work(struct w
 					#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
 					   &ctxt->chan_def,
 					#endif
+					#if (BUILD_OPENWRT >= 2410)
+					   NL80211_RADAR_CAC_FINISHED, GFP_KERNEL, 0);
+					#else
 					   NL80211_RADAR_CAC_FINISHED, GFP_KERNEL);
+					#endif
 	rwnx_send_apm_stop_cac_req(rwnx_hw, radar->cac_vif);
 	rwnx_chanctx_unlink(radar->cac_vif);
 
@@ -1499,7 +1503,11 @@ void rwnx_radar_cancel_cac(struct rwnx_r
 						#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 14, 0)
 						   &ctxt->chan_def,
 						#endif
+						#if (BUILD_OPENWRT >= 2410)
+						   NL80211_RADAR_CAC_ABORTED, GFP_KERNEL, 0);
+						#else
 						   NL80211_RADAR_CAC_ABORTED, GFP_KERNEL);
+						#endif
 		rwnx_chanctx_unlink(radar->cac_vif);
 	}
 
