--- /dev/null
+++ b/Makefile.am
@@ -0,0 +1,2 @@
+ACLOCAL_AMFLAGS = -I m4
+SUBDIRS = .
--- /dev/null
+++ b/libcharset/Makefile.am
@@ -0,0 +1 @@
+ACLOCAL_AMFLAGS = -I m4
--- a/libcharset/configure.ac
+++ b/libcharset/configure.ac
@@ -34,6 +34,8 @@
 
 dnl           checks for installer options
 
+gl_RELOCATABLE_LIBRARY
+
 dnl           checks for programs
 
 dnl           checks for UNIX variants that set DEFS
